keps/sig-architecture/3935-oldest-node-newest-control-plane/kep.yaml

```diff
@@ -0,0 +1,49 @@
+title: Support Oldest Node And Newest Control Plane
+kep-number: 3935
+authors:
+  - "@derekwaynecarr"
+  - "@liggitt"
+owning-sig: sig-architecture
+participating-sigs:
+  - sig-cluster-lifecycle
+  - sig-network
+  - sig-node
+status: provisional
+creation-date: 2023-04-03
+reviewers:
+  - "@soltysh" # sig-apps
+  - "@nckturner" # sig-cloud-provider
+  - "@ahg-g" # sig-scheduling
```

Comment:
There are three scheduler features related to this:
1. dynamic resource allocation
2. pod overhead 
3. in-place update of pod resources

Looking at the version skew strategy, it is not clear to me if the last one will be impacted (see https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#version-skew-strategy)

Reply:
I'm syncing with @vinaykul tomorrow on that, actually... that was the last outstanding question from the sig-node meeting

Reply:
Sounds good. 

This is captured in the KEP, but just as an FYI: we have shared logic between kubelet admission and scheduler filters. In the future, if we ever add a new filter that we also want to be part of kubelet admission, this skew policy change will delay the graduation of such feature.

Reply:
how do you coordinate enablement of new filters today with n-1 and n-2 skewed kubelets? do you enable the filters first in the kubelets, then wait for all supported skewed kubelets to have a filter enabled before enabling it in the scheduler?

Reply:
or do you just accept that older kubelets won't enforce that restriction and design the feature in a way that an older kubelet ignoring the filter is considered "safe"

Reply:
I don't think we went through this case recently. But depending on the feature, it will indeed be one of the two options yo mentioned above:

1) two feature flags, one guarding kubelet logic and starts in beta and one in the scheduler that starts in alpha and graduates to beta only after two (now three) releases.
2) kubelet admission is "best effort", we don't actually have all the filters as part of kubelet admission logic, only the simple ones are, and so a delay in enforcement by kubelet is considered safe.

Reply:
talked with @vinaykul about the in-place resize feature, and I don't think n-2 → n-3 skew impacts feature enablement there:

* a 1.27+ node with the feature disabled will not modify resources as requested, will mark pods requesting resize as "infeasible"
* a pre-1.27 node will not modify resources as requested, with no user feedback
* after 1.27 work, we realized that kubelet perpetually reports pod resize as InProgress when running against a containerd that supports UpdateContainerResources CRI API (containerd ~1.4/~1.5 era) but does not support ContainerStatus CRI API (added to CRI API in [k8s 1.25](https://github.com/kubernetes/kubernetes/pull/111645), supported in containerd 1.6.9+), so there's already user feedback improvements to make and possibly delay beta for


_if_ we were ready to promote in-place-resize to beta in 1.29, n-3 skew would mean 1.26 kubelets would not give any user feedback about lack of support for the feature, but would otherwise fail safe, so I don't think we would delay enablement _just_ for a nicer user experience when the feature wasn't enabled on a node